common/functions/src/scalars/dates/number_function.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::marker::PhantomData;
+
+use common_datavalues::arrays::DFUInt32Array;
+use common_datavalues::chrono::DateTime;
+use common_datavalues::chrono::Datelike;
+use common_datavalues::chrono::TimeZone;
+use common_datavalues::chrono::Utc;
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::Function;
+
+#[derive(Clone, Debug)]
+pub struct NumberFunction<T> {
+    display_name: String,
+    t: PhantomData<T>,
+}
+
+pub trait NumberResultFunction {
```

Comment:
Great, we can even make it more generic using `NumberResultFunction<T> `, this can be addressed in another pr to support ` toYYYYMMDDhhmmss`, which returns UInt64.

Reply:
I will address this in next pr.